tests/unit/test_tf_dataloader.py

```diff
@@ -272,3 +274,93 @@ def test_validater(tmpdir, batch_size):
     true_auc = roc_auc_score(labels, predictions)
     estimated_auc = logs[auc_key]
     assert np.isclose(true_auc, estimated_auc, rtol=1e-6)
+
+
+@pytest.mark.parametrize("engine", ["parquet"])
+@pytest.mark.parametrize("batch_size", [1, 10, 100])
+@pytest.mark.parametrize("global_rank", [0, 1])
+def test_multigpu_partitioning(datasets, engine, batch_size, global_rank):
+    cont_names = ["x", "y", "id"]
+    cat_names = ["name-string", "name-cat"]
+    label_name = ["label"]
+    data_loader = tf_dataloader.KerasSequenceLoader(
+        str(datasets["parquet"]),
+        cat_names=cat_names,
+        cont_names=cont_names,
+        batch_size=batch_size,
+        buffer_size=0.1,
+        label_names=label_name,
+        engine=engine,
+        shuffle=False,
+        global_size=2,
+        global_rank=global_rank,
+    )
+    indices = data_loader._gather_indices_for_dev(None)
+    assert indices == [global_rank]
+
+
+hvd = pytest.importorskip("horovod")
+
+
+def test_hvd(tmpdir):
+
+    json_sample = {
+        "conts": {},
+        "cats": {
+            "genres": {
+                "dtype": None,
+                "cardinality": 50,
+                "min_entry_size": 1,
+                "max_entry_size": 5,
+                "multi_min": 2,
+                "multi_max": 4,
+                "multi_avg": 3,
+            },
+            "movieId": {
+                "dtype": None,
+                "cardinality": 500,
+                "min_entry_size": 1,
+                "max_entry_size": 5,
+            },
+            "userId": {"dtype": None, "cardinality": 500, "min_entry_size": 1, "max_entry_size": 5},
+        },
+        "labels": {"rating": {"dtype": None, "cardinality": 2}},
+    }
+    cols = datagen._get_cols_from_schema(json_sample)
+    df_gen = datagen.DatasetGen(datagen.UniformDistro(), gpu_frac=0.0001)
```

Comment:
This is really neat, I didn't know we had this! (Makes sense that we do, but I hadn't seen it yet.)